bcp47-orphans/library/Data/BCP47/Serialise.hs

```diff
@@ -0,0 +1,23 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Data.BCP47.Serialise () where
+
+import Codec.Serialise
+import Codec.Serialise.Decoding
+import Codec.Serialise.Encoding
+import Control.Monad (void)
+import Data.BCP47 (BCP47, fromText, toText)
+import Data.Text (unpack)
+
+instance Serialise BCP47 where
+  -- bypass children not having their own Serialise instances by using toText/fromText
+  encode bcp = encodeListLen 2 <> encodeTag 0 <> encodeString (toText bcp)
+  decode = do
+    -- these two must be read to consume them
+    -- ignore them as they're here as part of the standard format but we don't need them
+    void decodeListLen
+    void decodeTag
```

Comment:
Wait. If you don't need them. Then why are you including them in the `encode`?

Can't you do,

```hs
encode = encodeString . toText
decode = either (fail . unpack) pure . fromText
```

?

Reply:
it's the format the package recommends, obviously we don't have to obey it, but figured i should follow their guidelines. it's intended to allow us compatibility should we ever expand the data type with new tags.

Reply:
> it's the format the package recommends

Can you say more?

> it's intended to allow us compatibility should we ever expand the data type with new tags

As long as our own `fromText`/`toText`s continues to round-trip, I don't see how us changing the data type would require changing such an instance. What am I missing?